### PR TITLE
stream: add exception policy for midstream flows [6.0.x backports] - v2

### PR DESCRIPTION
**Pull request by jufajardini**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
Exception policy midstream
- https://redmine.openinfosecfoundation.org/issues/5500
applayer parser sections merging:
- https://redmine.openinfosecfoundation.org/issues/5364
starts to address documenting the exception policy:
- https://redmine.openinfosecfoundation.org/issues/5475

Describe changes:
- add a stream.midstream-policy option that allows for controlling if a flow should be fully dropped, passed or bypassed. It is also possible to just ignore the policy (keeps normal behavior) if it is a midstream flow (in IPS mode)
- merge the two existing application layer parser configuration sections in the userguide into one
- add a very brief mention to each of the existing exception policy options (that are configurable via suricata.yaml) to the userguide

Previous PR: #7816 

Updates from last PR:
- bring documentation about HTTP2 settings to suricata.yaml userguide section
- remove section about maximum transactions, as this is only for master

suricata-verify-pr: 921
https://github.com/OISF/suricata-verify/pull/921

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 8981

**Comment by jufajardini** [comment]
CIFuzz failure is still around:
"Failed with input length 26 versus 260, found enip instead of smb
Assertion failure: enip-smb"

Shouldn't 7dc493cf55c6058 fix that?

File: doc/userguide/configuration/suricata-yaml.rst [hunk 1]
```diff
@@ -1572,6 +1601,118 @@ the limits are exceeded, and an event will be raised.
...
+
+      modbus:
+        # Stream reassembly size for modbus, default is 0
+        stream-depth: 0
+
+
+MQTT
+~~~~
+
+MQTT messages could theoretically be up to 256MB in size, potentially
```
Comment (jlucovsky) [inline doc/userguide/configuration/suricata-yaml.rst:1682]: Suggest "The maximum size of a MQTT message is 256MB, potentially ..."

File: doc/userguide/configuration/suricata-yaml.rst [hunk 2]
```diff
@@ -1572,6 +1601,118 @@ the limits are exceeded, and an event will be raised.
...
+containing a lot of payload data (such as properties, topics, or
+published payloads) that would end up parsed and logged. To acknowledge
+the fact that most MQTT messages, however, will be quite small and to
+reduce the potential for denial of service issues, it is possible to limit
+the maximum length of a message that we are willing to parse. Any message
```
Comment (jlucovsky) [inline doc/userguide/configuration/suricata-yaml.rst:1687]: s/we are willing to parse/Suricata should parse/

File: doc/userguide/configuration/suricata-yaml.rst [hunk 3]
```diff
@@ -1231,6 +1250,16 @@ network inspection.
 Application Layer Parsers
 -------------------------
 
+The ``app-layer`` section holds application layer specific configurations.
+
+A in IPS mode, a global exception policy accessed via the ``error-policy``
```
[the remark this answers is not in the document]
Comment (jufajardini) [reply doc/userguide/configuration/suricata-yaml.rst:1255]: 😅  that I re-read, after having more coffee, probably :P

**Comment by victorjulien** [comment]
Since this is a backport shall we fix things in master and backport that too?

**Comment by jufajardini** [comment]
> Since this is a backport shall we fix things in master and backport that too?

Yes sir! :)

File: doc/userguide/configuration/suricata-yaml.rst (hunk 2 above)
Comment (jufajardini) [reply doc/userguide/configuration/suricata-yaml.rst:1687]: > s/we are willing to parse/Suricata should parse/

As per Victor's suggestion, I've addressed all those in https://github.com/OISF/suricata/pull/7822 
thanks :D

**Comment by victorjulien** [comment]
Merged in #7828, thanks!

**Pull request closed without merging**